pkg/controller/managedcluster/managedcluster_import_detach.go

```diff
@@ -23,35 +23,69 @@ import (
 	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 
-	"github.com/open-cluster-management/library-go/pkg/applier"
-	"github.com/open-cluster-management/library-go/pkg/templateprocessor"
+	"github.com/open-cluster-management/applier/pkg/applier"
+	"github.com/open-cluster-management/applier/pkg/templateprocessor"
 )
 
 func (r *ReconcileManagedCluster) importCluster(
 	managedCluster *clusterv1.ManagedCluster,
+	clusterDeployment *hivev1.ClusterDeployment,
 	autoImportSecret *corev1.Secret) (res reconcile.Result, err error) {
 	res = reconcile.Result{}
-	var client client.Client
+
+	//Assuming that is a local import
+	client := r.client
```

Comment:
does that mean we will always import local? even for manual import scenario?

Reply:
The importCluster is called only if clusterDeployment or autoImportSecret present or managedCluster has the label for local-import.
https://github.com/itdove/managedcluster-import-controller/blob/05910be5a8ff93d4c3[…]216d/pkg/controller/managedcluster/managedcluster_controller.go
Line 292